jenkins/promotion/release-promotion.jenkinsfile

```diff
@@ -377,46 +428,92 @@ pipeline {
-            agent {
-                docker {
-                    label AGENT_LINUX_X64
-                    image 'docker/library/alpine:3'
-                    registryUrl 'https://public.ecr.aws/'
-                    alwaysPull true
-                }
+    stage('Validate the artifacts') {
+        agent {
+            docker {
+                label AGENT_LINUX_X64
+                image 'docker/library/alpine:3'
+                registryUrl 'https://public.ecr.aws/'
+                alwaysPull true
             }
-            steps {
-                echo 'Triggering distribution-promote-artifacts for OpenSearch Dashboards Linux tar x64'
-                build job: 'distribution-promote-artifacts', wait: true, parameters: [string(name: 'DISTRIBUTION_JOB_NAME', value: 'distribution-build-opensearch-dashboards'), 
-                                                                                    string(name: 'DISTRIBUTION_PLATFORM', value: 'linux'),
-                                                                                    string(name: 'DISTRIBUTION_NAME', value: 'tar'),
-                                                                                    string(name: 'DISTRIBUTION_ARCHITECTURE', value: 'x64'),
-                                                                                    string(name: 'DISTRIBUTION_BUILD_NUMBER', value: params.OPENSEARCH_DASHBOARDS_RC_BUILD_NUMBER),
-                                                                                    string(name: 'INPUT_MANIFEST', value: "${params.RELEASE_VERSION}/opensearch-dashboards-${params.RELEASE_VERSION}.yml"),
-                                                                                ]
-                echo 'Promotion successful for OpenSearch Dashboards Linux tar x64!'
+        }
+        steps {
+            echo 'Triggering validation workflow'
+            build job: 'distribution-validation', wait: true, parameters: [
+                string(name: 'VERSION', value: "${params.RELEASE_VERSION}"),
+                string(name: 'DISTRIBUTION', value: 'tar rpm yum deb zip docker'),
+                string(name: 'ARCHITECTURE', value: 'x64 arm64'),
+                string(name: 'PLATFORM', value: 'linux windows'),
+                string(name: 'PROJECTS', value: 'Both'),
+                string(name: 'ARTIFACT_TYPE', value: 'production'),
+                string(name: 'OPTIONAL_ARGS', value: 'validate-digests-only')
+                ]
+            echo 'Artifacts are successfully validated!'
+            }
+        }
+    stage('Publish to Maven'){
+        agent {
+            docker {
+                label AGENT_LINUX_X64
+                image 'docker/library/alpine:3'
+                registryUrl 'https://public.ecr.aws/'
+                alwaysPull true
+            }
+        }
+        steps {
+            echo 'Triggering publish to maven workflow'
+            build job: 'publish-to-maven', wait: true, parameters: [
```

Comment:
I think if the previous validation failed, then wait or error out the run.
Else, I have no concerns on straight follow up with maven release.

Reply:
It will error out. Tested that. The motivation to set `wait:false` is because maven takes a lot of time. And we do not want to wait in the release call for that. Let me know if that is okay.

Thanks!